### PR TITLE
Updated the use case layer for Login system

### PR DESCRIPTION
**Pull request by happynasit**
Description: Implemented methods for CreateUser, LoginUser, and delete user. Created classes and interfaces for each use case such that it aligns with Clean Architecture. Also, I implemented the UI of the login page and CreateAccount page.

- [ ]  Testing
- [ ]  Exception Handling
- [ ]  JavaDocs Documentation
- [ ]  SOLID Principles
- [ ]  Clean Architecture

### CODE REVIEW

File: src/main/java/entities/UserEntity.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        boolean b = false;
+        for (int i = 0; i < password.length(); i++){
+            if (Character.isDigit(password.charAt(i))){
+                b = true;
+            }
+        }
+        return b;
+    }
+
+    public boolean check_characters(){
```
Comment (JoAndNight) [inline src/main/java/entities/UserEntity.java]: Please add more JavaDoc.

File: src/main/java/use_cases/habits/edit_habit/EditHabitInputBoundary.java [hunk 2]
```diff
@@ -12,7 +12,7 @@ public class EditHabitInputBoundary {
      * @param t - The new type of the habit
      * @param r - The new reminder state of the habit
      */
-    public static void editHabitInputBoundary(User u, String s, String a, String t, Boolean r){
+    public static void editHabitInputBoundary(UserEntity u, String s, String a, String t, Boolean r){
```
Comment (JoAndNight) [inline src/main/java/use_cases/habits/edit_habit/EditHabitInputBoundary.java]: Code smell: long parameter list

File: src/main/java/use_cases/user/create_user/CreateUserInputData.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    /**
+     * Constructor for a CreateUserInputData object.
+     * @param emailaddress - the input email address
+     * @param password - the input password
+     * @param firstname - the input first name
+     * @param lastname - the input last name
+     * @param repeatpassword - repeat the password
+     */
+    public CreateUserInputData(String emailaddress, String password, String firstname, String lastname, String repeatpassword) {
```
Comment (JoAndNight) [inline src/main/java/use_cases/user/create_user/CreateUserInputData.java:20]: Code smell: long parameter list

File: src/main/java/use_cases/user/create_user/CreateUserInteractor.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    /**
+     * Create a new task based on input data.
+     * @param inputData - the input data which may contain the email address and password of the user.
+     * @return the output data after creation of new user.
+     */
+
+    @Override
+    public CreateUserOutputData create(CreateUserInputData inputData) {
```
Comment (JoAndNight) [inline src/main/java/use_cases/user/create_user/CreateUserInteractor.java:27]: Code smell: This method is too long.

File: src/main/java/use_cases/user/create_user/UserDataAccessInterface.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package use_cases.user.create_user;
+
+import entities.UserEntity;
+
+public interface UserDataAccessInterface {
```
Comment (JoAndNight) [inline src/main/java/use_cases/user/create_user/UserDataAccessInterface.java:5]: Add more JavaDoc

**Review by JoAndNight — APPROVED** [review]
Overall, you need to add more JavaDoc.

**Review by JoAndNight — APPROVED** [review]

File: src/main/java/use_cases/user/create_user/CreateUserDsRequestModel.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package use_cases.user.create_user;
+
+public class CreateUserDsRequestModel {
+
+    private final String emailaddress;
+    private String password;
+
+
```
Comment (MadhavKanna) [inline src/main/java/use_cases/user/create_user/CreateUserDsRequestModel.java:8]: Please add all parameters of the user or pass in the full created User object itself.

**Review by MadhavKanna — APPROVED** [review]

File: src/main/java/use_cases/user/delete_user/DeleteUserDsRequestModel.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
+package use_cases.user.delete_user;
+
+import entities.CommonUser;
+
+/**
+ * use case layer
+ */
```
Comment (AngelEyesMagicMaker) [inline src/main/java/use_cases/user/delete_user/DeleteUserDsRequestModel.java:7]: Can add some Javadoc to explain more.

**Review by AngelEyesMagicMaker — APPROVED** [review]

File: src/main/java/entities/UserEntity.java [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+        this.lastName = last_name;
+        this.habitCollection = new HabitFactory();
+        this.taskCollection = new TaskFactory();
+        this.categoryCollection = new CategoryFactory();
+        this.tomato = new Timer();
+    }
+
+    public UserEntity(){}
+
+    public UserEntity(String s, String s1, String harry, String potter, HabitFactory hFactory, TaskFactory tFactory, CategoryFactory cFactory, Timer t) {
```
[the remark this answers is not in the document]
Comment (happynasit) [reply src/main/java/entities/UserEntity.java]: This was created for an example class to test the code. It is removed from the entity.

File: src/main/java/entities/UserEntity.java (hunk 1 above)
Comment (happynasit) [reply src/main/java/entities/UserEntity.java]: Added the javadocs.

File: src/main/java/use_cases/user/create_user/UserDataAccessInterface.java (hunk 5 above)
Comment (happynasit) [reply src/main/java/use_cases/user/create_user/UserDataAccessInterface.java:5]: Added the Javadocs

**Pull request merged**